### PR TITLE
feat(icon, flag): updateComplete resolves after svg is ready

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

override getUpdateComplete to stabilize icon & flag element test

## Type of change

Please delete options that are not relevant.

- [x] Refactor (improves code without changing its functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by wattachai-lseg** [comment]
As of 11:10AM UTC 2023-05-15, there is an ongoing major outage of npm. https://status.npmjs.org/ The pipeline might fail unexpectedly.

File: packages/elements/src/flag/index.ts [hunk 1]
```diff
@@ -114,6 +116,19 @@ export class Flag extends BasicElement {
...
+   * A deferred promise representing flag ready.
+   * It would be resolved when the flag svg has been fetched and parsed, or
+   * when the flag svg is unavailable/invalid.
+   */
+  private flagReady!:Deferred<void>;
+
+  constructor () {
+    super();
+    this.flagReady = new Deferred<void>();
+    this.flagReady.resolve();
```
Comment (Nantawat-Poothong) [inline packages/elements/src/flag/index.ts:132]: How resolve flagReady in constructor difference from not resolve?

File: packages/utils/src/loader/deferred.ts [hunk 2]
```diff
@@ -12,6 +18,8 @@ export class Deferred<T> {
   private _reject: PromiseField<T> = () => {
   };
 
+  private state = PromiseState.pending;
```
Comment (Sarin-Udompanish) [inline packages/utils/src/loader/deferred.ts:17]: Would this better ?

```
  private _resolve!: (value?: T | PromiseField<T>) => void;
  private _reject!: (value?: T | PromiseField<T>) => void;
  private state = PromiseState.pending;
  private _promise: Promise<T>;

  constructor () {
    this._promise = new Promise<T>((resolve, reject) => {
      this._resolve = resolve as (value?: T | PromiseField<T>) => void;
      this._reject = reject;
    });
  }
```

File: packages/elements/src/flag/index.ts (hunk 1 above)
Comment (wattachai-lseg) [reply packages/elements/src/flag/index.ts:132]: Clarification comment added. Hope this clears things up ;)

File: packages/utils/src/loader/deferred.ts (hunk 2 above)
Comment (wattachai-lseg) [reply packages/utils/src/loader/deferred.ts:17]: Constructor looks too verbose for me. However, non-null assertion is quite helpful. Please check the updated version.

**Review by Sarin-Udompanish — APPROVED** [review]

**Review by Sarin-Udompanish — APPROVED** [review]

**Pull request merged**